easybuild/framework/easyblock.py

```diff
@@ -947,6 +947,10 @@ def make_module_dep(self, unload_info=None):
         self.log.debug("List of excluded deps: %s", excluded_deps)
 
         deps = [d for d in deps if d not in excluded_deps]
+        for dep in excluded_deps:
+            excluded_dep_deps = dependencies_for(dep, self.modules_tool)
```

Comment:
@bartoldeman While looking into a test for this, I ran into a problem...

This call to `dependencies_for` assumes that the module that corresponds with `dep` is *available* in the current environment, which it may not be.

For example, when including OpenMPI as a dependency in an easyconfig that uses the `goolf` toolchain, I run into something like this:

```
Traceback (most recent call last):
  File "/Users/kehoste/work/easybuild-framework/test/framework/easyblock.py", line 459, in test_make_module_dep_hmns
    print eb.make_module_dep()
  File "/Users/kehoste/work/easybuild-framework/easybuild/framework/easyblock.py", line 952, in make_module_dep
    excluded_dep_deps = dependencies_for(dep, self.modules_tool)
  File "/Users/kehoste/work/easybuild-framework/easybuild/tools/module_generator.py", line 734, in dependencies_for
    mod_filepath = modtool.modulefile_path(mod_name)
  File "/Users/kehoste/work/easybuild-framework/easybuild/tools/modules.py", line 570, in modulefile_path
    modpath = self.get_value_from_modulefile(mod_name, modpath_re)
  File "/Users/kehoste/work/easybuild-framework/easybuild/tools/modules.py", line 559, in get_value_from_modulefile
    raise EasyBuildError("Can't get value from a non-existing module %s", mod_name)
EasyBuildError: "Can't get value from a non-existing module OpenMPI/1.6.4"
```

This can be fixed, by loading all the modules that expand the module tree before this loop (basically, the modules for everything listed in `excluded_deps`).
That does make things slightly trickier though, since we have to make sure we also clean up after ourselves before we exit `make_module_dep`...